File: mvp-workflow-app/backend/services/workflow_service.py
from ..models.workflow import Workflow
from ..utils.workflow_executor import WorkflowExecutor
from typing import List, Optional
import uuid

class WorkflowService:
    def __init__(self):
        self.workflows = {}

    async def create_workflow(self, workflow: Workflow) -> Workflow:
        workflow_id = str(uuid.uuid4())
        self.workflows[workflow_id] = workflow
        
        # ワークフローを実行
        await WorkflowExecutor.execute_workflow(workflow)
        
        return workflow

    async def get_workflow(self, workflow_id: str) -> Optional[Workflow]:
        return self.workflows.get(workflow_id)

    async def list_workflows(self) -> List[Workflow]:
        return list(self.workflows.values())

    async def update_workflow(self, workflow_id: str, workflow: Workflow) -> Optional[Workflow]:
        if workflow_id in self.workflows:
            self.workflows[workflow_id] = workflow
            return workflow
        return None

    async def delete_workflow(self, workflow_id: str) -> bool:
        if workflow_id in self.workflows:
            del self.workflows[workflow_id]
            return True
        return False

File: mvp-workflow-app/backend/src/services/workflow_service.ts
import { Workflow } from "../models/workflow";
import { WorkflowExecutor } from "../utils/workflow_executor";
import { v4 as uuidv4 } from "uuid";

export class WorkflowService {
  private workflows: { [key: string]: Workflow } = {};

  async createWorkflow(workflow: Workflow): Promise<Workflow> {
    const workflowId = uuidv4();
    this.workflows[workflowId] = workflow;

    // ワークフローを実行
    await WorkflowExecutor.executeWorkflow(workflow);

    return workflow;
  }

  async getWorkflow(workflowId: string): Promise<Workflow | undefined> {
    return this.workflows[workflowId];
  }

  async listWorkflows(): Promise<Workflow[]> {
    return Object.values(this.workflows);
  }

  async updateWorkflow(
    workflowId: string,
    workflow: Workflow
  ): Promise<Workflow | undefined> {
    if (workflowId in this.workflows) {
      this.workflows[workflowId] = workflow;
      return workflow;
    }
    return undefined;
  }

  async deleteWorkflow(workflowId: string): Promise<boolean> {
    if (workflowId in this.workflows) {
      delete this.workflows[workflowId];
      return true;
    }
    return false;
  }
}
